examples/restrict_accessor/restrict_accessor.cpp

```diff
@@ -0,0 +1,298 @@
+  }
+}
+
+template<class ElementType>
+auto benchmark_add_mdspan_1d(const std::size_t num_trials, const index_type n, const ElementType* x, const ElementType* y, ElementType* z)
+{
+  TICK();
+  mdspan_1d<const ElementType> x2{x, n};
+  mdspan_1d<const ElementType> y2{y, n};
+  mdspan_1d<ElementType> z2{z, n};
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_mdspan_1d(x2, y2, z2);
+  }
+  return TOCK();
+}
+
+template<class ElementType>
+void add_raw_1d(const index_type n, const ElementType x[], const ElementType y[], ElementType z[])
+{
+  for (index_type i = 0; i < n; ++i) {
+    z[i] = x[i] + y[i];
+  }
+}
+
+template<class ElementType>
+auto benchmark_add_raw_1d(const std::size_t num_trials, const index_type n, const ElementType* x, const ElementType* y, ElementType* z)
+{
+  TICK();
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_raw_1d(n, x, y, z);
+  }
+  return TOCK();
+}
+
+template<class ElementType>
+void add_restrict_raw_1d(const index_type n,
+  _MDSPAN_RESTRICT_POINTER(const ElementType) x,
+  _MDSPAN_RESTRICT_POINTER(const ElementType) y,
+  _MDSPAN_RESTRICT_POINTER(ElementType) z)
+{
+  for (index_type i = 0; i < n; ++i) {
+    z[i] = x[i] + y[i];
+  }
+}
+
+template<class ElementType>
+auto benchmark_add_restrict_raw_1d(const std::size_t num_trials, const index_type n, const ElementType* x, const ElementType* y, ElementType* z)
+{
+  TICK();
+  _MDSPAN_RESTRICT_POINTER(const ElementType) x2 = x;
+  _MDSPAN_RESTRICT_POINTER(const ElementType) y2 = y;
+  _MDSPAN_RESTRICT_POINTER(ElementType) z2 = z;
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_restrict_raw_1d(n, x2, y2, z2);
+  }
+  return TOCK();
+}
+
+// Start with restrict pointers, then assign to nonrestrict pointers.
+// This checks whether the compiler's alias analysis suffices.
+template<class ElementType>
+void add_unrestrict_raw_1d(const index_type n,
+  _MDSPAN_RESTRICT_POINTER(const ElementType) x,
+  _MDSPAN_RESTRICT_POINTER(const ElementType) y,
+  _MDSPAN_RESTRICT_POINTER(ElementType) z)
+{
+  const ElementType* x2 = x;
+  const ElementType* y2 = y;
+  ElementType* z2 = z;
+  add_raw_1d(n, x2, y2, z2);
+}
+
+template<class ElementType>
+auto benchmark_add_unrestrict_raw_1d(const std::size_t num_trials, const index_type n, const ElementType* x, const ElementType* y, ElementType* z)
+{
+  TICK();
+  _MDSPAN_RESTRICT_POINTER(const ElementType) x2 = x;
+  _MDSPAN_RESTRICT_POINTER(const ElementType) y2 = y;
+  _MDSPAN_RESTRICT_POINTER(ElementType) z2 = z;
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_unrestrict_raw_1d(n, x2, y2, z2);
+  }
+  return TOCK();
+}
+
+template<class ElementType>
+void warmup(const index_type n, const ElementType* x, const ElementType* y, ElementType* z)
+{
+  constexpr std::size_t num_warmup_trials = 10;
+  for(std::size_t trial = 0; trial < num_warmup_trials; ++trial) {
+    for (index_type i = 0; i < n; ++i) {
+      z[i] = x[i] + y[i];
+    }
+  }
```

Comment:
Any good reason not to just the benchmarks directly to do the warmup?

Reply:
@dalg24 lol no ; - )

Reply:
@dalg24 fixed; thanks!